torch/distributed/fsdp/_shard_utils.py

```diff
@@ -124,3 +130,22 @@ def _create_chunk_sharded_tensor(
     return ShardedTensor._init_from_local_shards_and_global_metadata(
         local_shards, sharded_tensor_metadata=sharded_tensor_metadata, process_group=pg
     )
+
+
+def _create_chunk_dtensor(
+    tensor: torch.Tensor,
+    rank: int,
+    device_mesh: DeviceMesh,
+) -> DTensor:
+    """
+    Shard a tensor to chunks along the first dimension. The local rank will gets its
+    corresponding chunk as the local tensor to create a DTensor.
+    """
+    shard_placement = DShard(0)
+    tensor_padded_list, _ = shard_placement._split_tensor(
+        tensor,
+        device_mesh.size(dim=0),
+        with_padding=False,
+        contiguous=True,
+    )
+    return DTensor.from_local(tensor_padded_list[rank], device_mesh, [shard_placement])
```

Comment:
If this is for `state_dict` usage, I am wondering if we could call `from_local` under `torch.no_grad`, otherwise it would have a grad_fn associated with it, not sure if that matters much